game/src/common/mod.rs

```diff
@@ -410,3 +411,60 @@ pub fn checkbox_per_mode(
     }
     Widget::custom_row(filters)
 }
+
+/// Many game states fit a pattern of managing a single panel, handling mouseover events, and other
+/// interactions on the map. Implementing this instead of `State` reduces some boilerplate.
+pub trait SimpleState {
+    /// Called when something on the panel has been clicked.
+    fn on_click(&mut self, ctx: &mut EventCtx, app: &mut App, x: &str, panel: &Panel)
```

Comment:
Do we intentionally not have a default impl for `on_click`?

Reply:
There's no reasonable default (except maybe treating "close" as `Transition::Pop`, but that's a bit too much implicit convention). At minimum, a `SimpleState` only has a panel with different actions, and no real map-space handling. I updated the comments a bit to make the usage more clear.